src/cache.rs

```diff
@@ -486,15 +486,17 @@ impl Cache {
                     let wildcards = type_check(term, initial_ctxt.clone(), self)?;
                     self.update_state(file_id, EntryState::Typechecking);
                     self.wildcards.insert(file_id, wildcards);
-                }
 
-                if let Some(imports) = self.imports.get(&file_id).cloned() {
-                    for f in imports.into_iter() {
-                        self.typecheck(f, initial_ctxt)?;
+                    if let Some(imports) = self.imports.get(&file_id).cloned() {
+                        for f in imports.into_iter() {
+                            self.typecheck(f, initial_ctxt)?;
+                        }
                     }
-                }
 
-                self.update_state(file_id, EntryState::Typechecked);
+                    self.update_state(file_id, EntryState::Typechecked);
+                }
+                // The else case correponds to `EntryState::Typechecking`. There is nothing to do:
+                // cf (grep for) [transitory_entry_state]
```

Comment:
This might just be my ignorance of the typechecking algorithm, but how does Nickel typecheck a file before having typecheckt its imports?